src/CoreBundle/private/library/classes/dbobjects/CronJobs/TCMSCronJob.class.php

```diff
@@ -179,41 +233,31 @@ protected function UpdateLastExecutionOnStart()
      */
     protected function _NeedExecution()
     {
-        $needExecution = false;
-        if (!empty($this->sqlData['last_execution'])) {
-            $iExecutionIntervalInSeconds = $this->sqlData['execute_every_n_minutes'] * 60;
-            $iLastExecutionTimeInSeconds = strtotime($this->sqlData['last_execution']);
-            $iNextExecutionTime = $iLastExecutionTimeInSeconds + (60 * $this->sqlData['execute_every_n_minutes']) - 30; // add a 30 second tolerance
-
-            if ('1' == $this->sqlData['lock']) {
-                // check if cronjob is locked, but last execution is older than 24h and older than execute interval
-                // may be possible if the cronjob was interrupted before end of script by an error or server time-out
-                if ($iNextExecutionTime <= time()) {
-                    $iMaxLockTime = $this->sqlData['unlock_after_n_minutes'];
-                    if (!empty($iMaxLockTime)) {
-                        $iMaxLockTime = 60 * $iMaxLockTime;
-                        if (time() - $iLastExecutionTimeInSeconds - $iExecutionIntervalInSeconds > $iMaxLockTime) {
-                            $needExecution = true;
-                        }
-                    } else {
-                        if ($iLastExecutionTimeInSeconds + $iExecutionIntervalInSeconds <= strtotime('-1 day')) {
-                            $needExecution = true;
-                        }
-                    }
-                    if ($needExecution) {
-                        $this->_Unlock();
-                    }
-                }
-            } else {
-                if ($iNextExecutionTime <= time()) {
-                    $needExecution = true;
-                }
-            }
-        } else {
-            $needExecution = true;
+        $scheduler = $this->getCronJobScheduler();
+        try {
+            $schedule = $this->getSchedule();
+        } catch (InvalidArgumentException $e) {
+            $this->getLogger()->error($e->getMessage(), __FILE__, __LINE__);
+
+            return false;
         }
 
-        return $needExecution;
+        $requiresExecution = $scheduler->requiresExecution($schedule);
+        if ($requiresExecution && $this->isLocked()) {
+            $this->getLogger()->warning(
+                sprintf(
+                    'CronJob "%s" (%s) was force unlocked due to it being locked for longer than its unlock_after_n_minutes value',
```

Comment:
```suggestion
                    'Cron job "%s" (%s) was unlocked due to it being locked for longer than its theshold.',
```

Reply:
since this is an exception I think it is ok to generate a technical error message. So using the field name makes sense to me. I did reword "CronJob" to "Cron job" though